### PR TITLE
v01.0166 Fixed bouncing speed in turn. Fixes #3312

### LINKED ISSUE #3312: AI Gas pedal "pulsation" on turn maneuvers
The AI pulsates the gas while on turn maneuver, rather than keeping a steady speed.
I'm not sure if this has already been brought up yet, but it does seem an obvious issue, although searching "turn" or "maneuver" didn't return any relevant issues.
Do you need any logs or is this a known issue?

### PR DESCRIPTION
**Pull request by pops64**
This requires using the driveInDirection function. The driveToPoint function has interal limits that kick back the speed when steering angle is great. diriveInDirection allows us to set when the kickback occurs

### CODE REVIEW

**Comment by GeorgeEarslight** [comment]
Works as intended, thanks!
#3312

**Comment by pops64** [comment]
No problem was an easy fix. Will have to see if CP team wants to include it. According to giants they no longer use driveInDirection so there was a push to switch to point. As driveInDirection is currently bugged and we have to completely overwrite there function to use it still.

File: turn.lua [hunk 1]
```diff
@@ -818,8 +818,10 @@ function courseplay:turn(vehicle, dt)
 
 		AIVehicleUtil.driveInDirection(vehicle, dt, vehicle.cp.steeringAngle, directionForce, 0.5, 20, allowedToDrive, moveForwards, lx, lz, refSpeed, 1);
 	else
-		dtpZ = dtpZ * 0.85;
-		AIVehicleUtil.driveToPoint(vehicle, dt, directionForce, allowedToDrive, moveForwards, dtpX, dtpZ, refSpeed);
+		-- This code causing the bouncing of the speeds during turns. We must use driveInDirection here to prevent this. As this Giants Function lets us defeine the slow angle limit where as point is a set calc
+		--dtpZ = dtpZ * 0.85;
+		--AIVehicleUtil.driveToPoint(vehicle, dt, directionForce, allowedToDrive, moveForwards, dtpX, dtpZ, refSpeed);
+		AIVehicleUtil.driveInDirection(vehicle, dt, vehicle.cp.steeringAngle, directionForce, 0.5, 20, allowedToDrive, moveForwards, lx, lz, refSpeed, 1);
```
Comment (pvaiko) [inline turn.lua]: This is exactly the same call as above, so there's no need for this else at all. Also we don't want commented out code in here, if it is not needed, remove it.

**Comment by Mrsidders** [comment]
I have updated the courseplay that was issued a few hrs ago with the bouncing turn fix, now when the machinery goes to make a turn it raises implement then speeds off without a turn, Im currently doing two headland passes

**Comment by pops64** [comment]
Which tool?

**Comment by Mrsidders** [comment]
When sowing

**Comment by Mrsidders** [comment]
2019-04-07 15:30 Error: Not all prerequisites of specialization workArea are fulfilled
2019-04-07 15:30 Error: Not all prerequisites of specialization fillVolume are fulfilled
2019-04-07 15:30 Error: Not all prerequisites of specialization sowingMachine are fulfilled
2019-04-07 15:30 Error: Not all prerequisites of specialization sprayer are fulfilled
2019-04-07 15:30   =C:/Users/MrSid/Documents/My Games/FarmingSimulator2019/mods/zzz_Courseplay/register.lua (97) : superFunc
2019-04-07 15:30   dataS/scripts/i3d/I3DManager.lua (118) : asyncCallbackFunction
2019-04-07 15:30   Error: Index not found: 0>0|41
2019-04-07 15:30   Error: Failed to find child 41 from node db90, only 40 childs given

**Comment by pvaiko** [comment]
@Mrsidders open an issue for this and make sure to include the entire log, although this looks like a mod conflict. Also, when we ask 'which tool' we expect that you tell us exactly which tool it is. Believe it or not, 'sowing' does not help.

**Comment by Mrsidders** [comment]
Well  i'm sorry I didn't give specifics as you can see it is the db90 sowing implement. I shall just delete this implement and try another

**Pull request merged**